src/main/java/org/embulk/input/zendesk/services/ZendeskNormalServices.java

```diff
@@ -131,18 +164,17 @@ private void importDataForIncremental(final ZendeskInputPlugin.PluginTask task,
 
                 // https://developer.zendesk.com/rest_api/docs/support/incremental_export#pagination
                 // When there are more than 1000 records share the same time stamp, the count > 1000
-                long apiEndTime = result.get(ZendeskConstants.Field.END_TIME).asLong();
                 startTime = startTime == apiEndTime
                         ? apiEndTime + 1
                         : apiEndTime;
 
-                if (numberOfRecords < ZendeskConstants.Misc.MAXIMUM_RECORDS_INCREMENTAL) {
+                if (numberOfRecords < ZendeskConstants.Misc.MAXIMUM_RECORDS_INCREMENTAL || apiEndTime > endTime) {
```

Comment:
Should we use `startTime > endTime` in this case, because we can waste 1 request if we hit the scenario where at `end_date` there are more than 1000 records have been uploaded by batch and next start time is `end_time + 1:

```
startTime = startTime == apiEndTime
                        ? apiEndTime + 1
                        : apiEndTime;
```

Reply:
ah you are right. That should be better

Reply:
fixed